### PR TITLE
feat(pip_setup): copy extract.py to repo

### LINKED ISSUE #6766: Running renovate/renovate:slim CLI with binary-source docker will fail for pip extractions
**What Renovate type are you using?**

Renovate CLI `renovate/renovate:slim` version `21.31.2` on Enterprise Github

**Describe the bug**

When running the dockerized CLI (`renovate/renovate:slim`) with mounted docker socket `-v /var/run/docker.sock:/var/run/docker.sock` and `--binary-source docker` Renovate will pull in the `renovate/pip` image from dockerhub to extract dependencies inside e.g. `setup.py`.

This will result in the executed command: 
```
DEBUG: python command (repository=my/repository)
       "cmd": "docker",
       "args": [
         "run",
         "-i",
         "--rm",
         "-v",
         "/var/jenkins/renovate/repos/github/my/repository:/var/jenkins/renovate/repos/github/my/repository",
         "-v",
         "/usr/src/app/node_modules/renovate/data/extract.py:/usr/src/app/node_modules/renovate/data/extract.py",
         "-w",
         "/var/jenkins/renovate/repos/github/my/repository",
         "renovate/pip",
         "python",
         "\"/usr/src/app/node_modules/renovate/data/extract.py\"",
         "\"setup.py\""
       ]
```

The problem is that the path to the `extract.py` is absolute in the view of the container, but is (most probably) not present on the host.
Docker will happily map the path to `extract.py` but its not a path to a script, it is a directory in that case.

This will lead to the following output:

**Relevant debug logs**

```
WARN: Failed to read setup.py file (repository=my/repository, packageFile=setup.py)
       "err": {
         "killed": false,
         "code": 1,
         "signal": null,
         "cmd": "docker run -i --rm -v /var/jenkins/renovate/repos/github/my/repository:/var/jenkins/renovate/repos/github/my/repository -v /usr/src/app/node_modules/renovate/data/extract.py:/usr/src/app/node_modules/renovate/data/extract.py -w /var/jenkins/renovate/repos/github/my/repository renovate/pip python \"/usr/src/app/node_modules/renovate/data/extract.py\" \"setup.py\"",
         "stdout": "",
         "stderr": "/usr/bin/python: can't find '__main__' module in '/usr/src/app/node_modules/renovate/data/extract.py'\n",
         "message": "Command failed: docker run -i --rm -v /var/jenkins/renovate/repos/github/my/repository:/var/jenkins/renovate/repos/github/my/repository -v /usr/src/app/node_modules/renovate/data/extract.py:/usr/src/app/node_modules/renovate/data/extract.py -w /var/jenkins/renovate/repos/github/my/repository renovate/pip python \"/usr/src/app/node_modules/renovate/data/extract.py\" \"setup.py\"\n/usr/bin/python: can't find '__main__' module in '/usr/src/app/node_modules/renovate/data/extract.py'\n",
         "stack": "Error: Command failed: docker run -i --rm -v /var/jenkins/renovate/repos/github/my/repository:/var/jenkins/renovate/repos/github/my/repository -v /usr/src/app/node_modules/renovate/data/extract.py:/usr/src/app/node_modules/renovate/data/extract.py -w /var/jenkins/renovate/repos/github/my/repository renovate/pip python \"/usr/src/app/node_modules/renovate/data/extract.py\" \"setup.py\"\n/usr/bin/python: can't find '__main__' module in '/usr/src/app/node_modules/renovate/data/extract.py'\n\n    at ChildProcess.exithandler (child_process.js:303:12)\n    at ChildProcess.emit (events.js:315:20)\n    at ChildProcess.EventEmitter.emit (domain.js:482:12)\n    at maybeClose (internal/child_process.js:1021:16)\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5)"
       },
       "content": "[content]"
```

**To Reproduce**

This can be reproduced by running the `renovate/renovate:slim` docker image on a repository with an arbitrary `setup.py`

**Additional context**

I currently fixed this by creating the path first and mounting it under a different name into the container, copying the scripts to the folder and then running renovate.

```
# on the host

# also prepare a folder on the host so that renovate can call docker scripts
mkdir -p /usr/src/app/node_modules/renovate/data
# and let it own by us
sudo chown $USER_GROUP /usr/src/app/node_modules/renovate/data
...
# mounting with volume
-v /usr/src/app/node_modules/renovate/data:/helpers
...
# in the container
cp /usr/src/app/node_modules/renovate/data/* /helpers/

```

Ideally the small script would be put into the `renovate/pip` image during building the docker image.
Another less ideal way would be to serialize the script and pass it to the `renovate/pip` docker command, so that you are not dependent on the filesystem.

### PR DESCRIPTION
**Pull request by viceice**
* copy `extract.py` to repo and call from there
* write extract result to file instead of `stdout`

closes: #6766

### CODE REVIEW

**Comment by viceice** [comment]
tested against real repo with and without docker. 🙃

File: data/extract.py [hunk 1]
```diff
@@ -25,6 +25,7 @@ def invoke(mock1, mock2):
   # called arguments are in `mock_setup.call_args`
   call_args = mock1.call_args or mock2.call_args
   args, kwargs = call_args
-  print(json.dumps(kwargs, indent=2))
+  with open('renovate-python-report.json', 'w', encoding='utf-8') as f:
+    json.dump(kwargs, f, ensure_ascii=False, indent=2)
```
Comment (JamieMagee) [inline data/extract.py:29]: Do we need to clean this up afterwards?

**Review by JamieMagee — DISMISSED** [review]

File: data/extract.py (hunk 1 above)
Comment (viceice) [reply data/extract.py:29]: Don't think so, we do the same with gradle (no cleanup)

**Comment by rarkins** [comment]
Do we need to make the file name sufficiently unique?

**Comment by viceice** [comment]
Adapted from gradle manager, maybe it's better to write to `cacheDir` to not interferre with repo? or to `<localDir>/.git/renovate` ?

**Comment by viceice** [comment]
I think this is fine now. We will hopefully have a js based manager soon. 🙃

**Review by rarkins — APPROVED** [review]
LGTM assuming it's been tested both with and without Docker

**Review by rarkins — APPROVED** [review]

**Pull request merged**